bst.py

```diff
@@ -1,61 +1,89 @@
+            return self.left.depth
+        else:
+            return 0
+
+    def right_depth(self):
+        if self.right:
+            return self.right.depth
+        else:
+            return 0
+
+    def _update_depth(self):
+        self.depth = max(self.left_depth(), self.right_depth()) + 1
+
+    def insert(self, value_to_insert):
+        if self.value == value_to_insert:  # not allowing duplicates
             return False
-        elif self.value > data:
-            if self.leftChild:
-                return self.leftChild.insert(data)
-            else:
-                self.leftChild = Node(data)
-                return True
         else:
-            if self.rightChild:
-                return self.rightChild.insert(data)
+            return_val = False
+            if self.value > value_to_insert:
+                if self.left:
+                    return_val = self.left.insert(value_to_insert)
+                else:
+                    self.left = _BstNode(value_to_insert)
+                    return_val = True
             else:
-                self.rightChild = Node(data)
-                return True
+                if self.right:
+                    return_val = self.right.insert(value_to_insert)
+                else:
+                    self.right = _BstNode(value_to_insert)
+                    return_val = True
+            self._update_depth()
+            return return_val
 
-    def contains(self, data):
-        if (self.value == data):
+    def contains(self, value_to_find):
+        if self.value == value_to_find:
             return True
-        elif self.value > data:
-            if self.leftChild:
-                return self.leftChild.contains(data)
+        elif self.value > value_to_find:
+            if self.left:
+                return self.left.contains(value_to_find)
             else:
                 return False
         else:
-            if self.rightChild:
-                return self.rightChild.contains(data)
+            if self.right:
+                return self.right.contains(value_to_find)
             else:
                 return False
 
 
-class Tree:
+class BST(object):
     def __init__(self):
-        self.root = None
+        self._size = 0
+        self._root = None
+        self._balance = 0
 
-    def length(self):
-        return self.size
+    def __len__(self):
+        return self._size
 
-    def insert(self, data):
+    def insert(self, val):
         """
         Insert the value val into the BST.  If val is already present,
         it will be ignored.
         """
-        if self.root:
-            return self.root.insert(data)
+
+        if self._root:
+            size_increased = self._root.insert(val)  # value to insert
+            if size_increased:
+                self._size += 1
+            return size_increased
         else:
-            self.root = Node(data)
+            self._root = _BstNode(val)
+            self._size += 1
+            return True
```

Comment:
if this says `size_increased = True` instead of `return True`, then you can remove the return statement above and collapse down to just one return in this method.  That'd be nice.